### PR TITLE
doc: contribution guidelines: Specify roles and responsibilies

### PR DESCRIPTION
**Pull request by dleach02**
Clarify the roles and responsibilities of the Zephyr community
and contributors with respect to PRs, Bugs, and Features.

Signed-off-by: David Leach <david.leach@nxp.com>

As usual, the "live" output of this change is here: https://builds.zephyrproject.org/zephyrproject-rtos/zephyr/19498/docs/contribute/index.html (scroll to the bottom)

### CODE REVIEW

**Comment by dleach02** [comment]
This is an initial pass of roles and responsibilities for community and contributors. The final bit on this is the "timed duration" we would put on feedback from contributors. I'm posting this to initiate discussion.

**Review by aescolar — CHANGES_REQUESTED** [review]
Thanks @dleach02,
if this stems from the TSC action you took for noting 
in the documentation how long we would wait before closing an issue or PR,
then it would seem the last section alone would be enough.

Regarding that particular last paragraph, I'm quite ok with what you propose,
I could bikeshed saying that there is no need to over-specify a 2nd attempt
and extra week, as anyhow it would really be at the discretion of the
maintainer doing it, and would normally take longer than just 1 week for
a maintainer to close it.

About all the text you added before that,
it would seem that you are only trying to convey one real message there(?):
That a contributor is expected to 
"provide timely responses, and work with other community members
to resolve the issue at hand", be it bringing a PR to maturity or
helping debug an issue.
If so, I think it would be better to resume all that in a small
paragraph. As it stands here it is overly verbose and fuzzy.

**Comment by dleach02** [comment]
@aescolar, I actually had two objectives. One being what came out of the action item to put a time bounding on contributor responses but the other was to highlight or outline what the expectations can be of the community (although there is no time bounding on the community). I didn't really want to make it a one sided affair. 

I would like to keep something about the requirements on the community. I could simplify things down that the community is expected to review all PRs/bug reports/feature requests in a timely manner and the contributor is expected to provide timely responses to any engagement on their submission (instead of breaking this down into the three categories). Lack of timely response from contributor can result in the submission being closed without further action.

**Comment by aescolar** [comment]
@dleach02 I'd go with just setting the expectations for the contributor regarding PRs or issues being closed in 10 days without response as was agreed in the TSC (as this is a change in the contributor guidelines).
Better to fix one thing at a time.

About writing here requirements for the community. Just writing that is not going to help, but it may, on the other hand, set unrealistic expectations on contributors.

File: doc/contribute/index.rst
Comment (nashif) [inline doc/contribute/index.rst]: spell it out, Pull Requests

File: doc/contribute/index.rst
Comment (nashif) [inline doc/contribute/index.rst]: or Draft

File: doc/contribute/index.rst
Comment (nashif) [inline doc/contribute/index.rst]: There lots of documentation regarding this in doc/development_process/dev_env_and_tools.rst

I suggest to either combine or make sure we do not duplicate.

**Review by nashif — CHANGES_REQUESTED** [review]

File: doc/contribute/index.rst
Comment (dleach02) [reply doc/contribute/index.rst]: Okay. I'll review that section as well.

**Review by marc-hb — COMMENTED** [review]
Please mention this doesn't apply to github drafts. A bit obvious but better safe than sorry.

File: doc/contribute/index.rst
Comment (marc-hb) [reply doc/contribute/index.rst]: This entire paragraph (and review comment) are gone in the latest force push.

Comment (dleach02) [reply doc/contribute/index.rst]: yes, it was pointed out that there is related content in the other file I updated so I modified things (plus there was an objection for me explicitly calling out roles).

**Comment by dleach02** [comment]
> Please mention this doesn't apply to github drafts. A bit obvious but better safe than sorry.

Yes, will do. I accidentally dropped that part with the rework.

**Comment by alexanderwachter** [comment]
@nashif @MaureenHelm Is there a document where we define maintainers?
Wouldn't it be good to have a list of maintainers somewhere?

**Comment by marc-hb** [comment]
> @nashif @MaureenHelm Is there a document where we define maintainers?

Have a look at recent TSC minutes, I think there's some effort now to formalize this.

> Wouldn't it be good to have a list of maintainers somewhere?

CODEOWNERS?

**Review by marc-hb — DISMISSED** [review]
As previously mentioned by @nashif - elaborating and reposting at the top-level to dodge force pushes: there's another section about code review and time in a different place:
https://docs.zephyrproject.org/latest/development_process/dev_env_and_tools.html#code-review

All code review time limits should be in the same place. Not sure which one is best... @dbkinder ?

Ideally, the entire code review process should be in the same place. There seems to be some duplication/divergence about "roles and responsibilities"  too.

File: doc/contribute/index.rst
Comment (aescolar) [inline doc/contribute/index.rst]: DNR => DNM

File: doc/development_process/dev_env_and_tools.rst
Comment (aescolar) [inline doc/development_process/dev_env_and_tools.rst]: > at the discretion of the appropriate working group

Bike-shedding a bit: most work is done by maintainers or their own, not workgroups. So I'd say it would be better to say "at the discretion of the involved maintainers" which is anyway generic enough (I really hope you did not mean to propose we have a meeting each time we need to close a PR or issue :) )

Comment (dleach02) [reply doc/development_process/dev_env_and_tools.rst]: We already have a meeting that reviews all the PR and bugs and we propose changes on these items. On Tuesdays. 

But yes, I was intending this to be the working group to do this and not necessarily an individual maintainer. The thinking here was to have a tool/filter that could identify these stale PR/Issues and have them marked stale or ready to close. The meeting could quickly go through them and make decision on close or not close. Note that this only occurs if there is a specific request for additional information and the contributor is not responding.

File: doc/contribute/index.rst
Comment (dleach02) [reply doc/contribute/index.rst]: Ok

File: doc/contribute/index.rst
Comment (dbkinder) [inline doc/contribute/index.rst]: We can remove the duplicated material here (now that you've added it to
dev_env_and_tools.rst) and simplify this section with this rewrite:

```
The Zephyr project defines a development process workflow using
GitHub **Issues** to track feature, enhancement, and bug reports
together with GitHub **Pull Requests** (PRs) for submitting and
reviewing changes.  Zephyr community members work together
with `project working groups
<https://wiki.zephyrproject.org/Zephyr-Committee-and-Working-Group-Meetings>`_
to review these Issues and PRs, managing feature enhancements and
quality improvements of Zephyr through its regular releases, as outlined
in this `program management overview
<https://wiki.zephyrproject.org/Program-Management>`_.

We can only manage the volume of Issues and PRs, by requiring
timely reviews, feedback, and responses from the reviewing working groups,
area maintainers, and contributors, both for initial submissions
and for followup questions and clarifications.  Read about the
project's :ref:`development processes and tools
<dev-environment-and-tools>` and specifics about 
:ref:`review timelines <review_time>` to learn about the project's
goals and guidelines for our active developer community.
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/development_process/dev_env_and_tools.rst
Comment (dleach02) [reply doc/development_process/dev_env_and_tools.rst]: agreed

File: doc/contribute/index.rst
Comment (dleach02) [reply doc/contribute/index.rst]: agreed

**Review by dbkinder — APPROVED** [review]
LGTM, thanks!

File: doc/development_process/dev_env_and_tools.rst
Comment (aescolar) [reply doc/development_process/dev_env_and_tools.rst]: > We already have a meeting that reviews all the PR and bugs and we propose changes on these items. On Tuesdays.

How things are nowadays, the "bug triage and release readiness meeting" (I presume that is the one you refer to) does certainly not cover all open PRs and issues.
Only a very small part of the open issues, and only the ones that are not yet fully triaged, are glimpsed into, and only for the purpose of quickly assigning them to somebody and stamping a priority on them. In that meeting, they are certainly not "reviewed".
Apart from this, this meeting is not a working group forum. It is just the release manager and "whoever shows up" for that meeting. And in general that group is not, at all, the same set of people which is reviewing the open set of PRs.

> I was intending this to be the working group to do this and not necessarily an individual maintainer. 

So what you are proposing is to change the scope of that meeting to be "bug triage, release readiness AND closing of stale PRs and issues"..
Ok.. I guess you could try to..
(defining this here seems quite far from the action you got to note that after 10 days without response issues are closed)

Anyhow, again, how this paragraph is phrased*, it gives the impression that, in general, PRs and issues are reviewed by "working groups" which is not at all how things are, or could be in the foreseeable future.

*Together with the line "Zephyr community members work together with project working groups
to review these Issues and PRs"

Comment (aescolar) [reply doc/development_process/dev_env_and_tools.rst]: If you just replace "After one more week without a response the item may be closed at the discretion of the appropriate working group reviewing the item (draft and DNM tagged pull requests are excluded)." 
with 
"After one more week without a response the item may be closed (draft and DNM tagged pull requests are excluded)." 
or with 
"After one more week without a response the item may be closed by the relevant maintainers (draft and DNM tagged pull requests are excluded)." 
I'd be ok with it.
("Relevant maintainers" is generic and vague enough, it can be a set of people in a working group, whoever was in a "closing stale issues and PRs" meeting, or the area maintainer(s) )

File: doc/contribute/index.rst
Comment (aescolar) [inline doc/contribute/index.rst]: `project working groups`, this is not a list of working groups, but of (most of all) forums.. In general many of them are not "working groups" with any authority.
Miss-understanding that leads to situations where contributors are led to run in circles until they give up in frustration.

I think the easiest way forward would be just to remove this paragraph and leave only the paragraph below

**Review by aescolar — CHANGES_REQUESTED** [review]

**Review by aescolar — CHANGES_REQUESTED** [review]
(The overall review comment was lost)
I hope you do not perceived my review as me trying to block this PR.
I understand you are, with good intentions, trying to redefine how things should be.
But the thing is, adding yet another set of paragraphs to the documentation which are not coherent with other parts of the documentation, or with how things are, is just going to add to the confusion.

File: doc/contribute/index.rst
Comment (dleach02) [reply doc/contribute/index.rst]: @aescolar, I believe your primary issue is using the concept of "working groups" so I'm rewording things a bit to just say "community" where to me it is implied that community means owners, maintainers, working groups, etc... I suspect that we will likely want to clarify that a bit more to formalize some aspects of the process for clarity reasons... particularly how it relates to PRs.

**Review by nashif — APPROVED** [review]

**Comment by dleach02** [comment]
@aescolar, you provided the most feedback so I would like your approval before I push to get this PR merged and closed.

**Review by aescolar — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**
